pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -193,6 +199,19 @@ func (svmc *SVMController) sync(ctx context.Context, key string) error {
 	// working with a copy to avoid race condition between this and resource version controller
 	toBeProcessedSVM := svm.DeepCopy()
 
+	gvr := getGVRFromResource(toBeProcessedSVM)
+
+	isUpdatable, err := svmc.isResourceUpdatable(gvr)
+	if err != nil {
+		return err
```

Comment:
This feels like a terminal error as well?

Reply:
I don't think it necessarily is, in the case before if it doesn't exist in discovery it should be handled by the previous error handler. If the discovery client is unavailable or something along those lines however then I think that should just be a retry.